### PR TITLE
add pester build function

### PR DESCRIPTION
**Pull request by raviduddela**
Previous versions were shipping binary.
Latest versions needs to be built with dotnet.
Added build function for pester.

### CODE REVIEW

File: packages/development/tools/misc/pester/plan.ps1 [hunk 1]
```diff
@@ -8,7 +8,11 @@ $pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
...
 $pkg_shasum="39dac4c4c19ad2ffce1e863bea8b68b05a648156833933ecde1826a49ac8f2d6"
 $pkg_bin_dirs=@("module/bin")
-
+$pkg_build_deps=@("core/dotnet-core-sdk")
+function Invoke-Build {
+	Set-Location "$pkg_name-$pkg_version"
+	.\build.ps1 -Clean
+}
 function Invoke-Install {
     Copy-Item "pester-$pkg_version\*" "$pkg_prefix/module" -Recurse -Force
```
Comment (mwrock) [inline packages/development/tools/misc/pester/plan.ps1:17]: The entire pester repo is landing in the module directory.
I think you want this to be:
```
Copy-Item "pester-$pkg_version\bin\*" "$pkg_prefix/module" -Recurse -Force
```

**Review by mwrock — CHANGES_REQUESTED** [review]
In core-plans there was a readme file that gave instructions on how to use this but it is missing here.

File: packages/development/tools/misc/pester/plan.ps1 (hunk 1 above)
Comment (raviduddela) [reply packages/development/tools/misc/pester/plan.ps1]: done

**Review by mwrock — APPROVED** [review]
Looks good!

**Pull request merged**